src/server/index.spec.js

```diff
@@ -0,0 +1,70 @@
+import path from 'path';
+import request from 'supertest';
+
+import { createServer, injectAuthenticationToken } from '.';
+
+describe(__filename, () => {
+  describe('injectAuthenticationToken', () => {
+    it('returns HTML with the token in it', () => {
+      const token = '123';
+
+      const req = {
+        universalCookies: {
+          get: () => token,
+        },
+      };
+      const env = { REACT_APP_AUTH_TOKEN_PLACEHOLDER: '__PLACEHOLDER__' };
+      const html = `<div data-token="${env.REACT_APP_AUTH_TOKEN_PLACEHOLDER}">`;
+
+      const htmlWithToken = injectAuthenticationToken(req, html, env);
+
+      expect(htmlWithToken).toEqual(`<div data-token="${token}">`);
+    });
+  });
+
+  describe('createServer', () => {
+    const rootPath = path.join(__dirname, 'fixtures');
+
+    describe('NODE_ENV=production', () => {
+      const env = {
+        NODE_ENV: 'production',
+      };
+
+      it('creates an Express server', async () => {
+        const server = request(createServer({ env, rootPath }));
```

Comment:
Each time supertest creates a server it gets bound to an ephemeral port. That's about all the docs say. I'm pretty sure that awaiting the promise will shut down the server but I'm not totally sure. We need to shut the server down after each test. 

On the frontend we used `.end()` to do this but in this newer version of supertest, it prints a warning saying that you shouldn't mix that with promises. If you have any ideas, let me know, but I guess we can just ride with it.

Reply:
> I'm pretty sure that awaiting the promise will shut down the server but I'm not totally sure. We need to shut the server down after each test.

Reading the docs, it is done, no? I can't find leaky processes.

Reply:
Yes, I think it's working